app/controllers/declaration/ProcedureCodesController.scala

```diff
@@ -126,15 +126,27 @@ class ProcedureCodesController @Inject()(
   private def updateCache(itemId: String, updatedProcedureCodes: ProcedureCodesData)(
     implicit r: JourneyRequest[AnyContent]
   ): Future[Option[ExportsDeclaration]] = {
+
+    def clearDataForProcedureCode(code: String, itemId: String, model: ExportsDeclaration) = {
+      val step1Model =
+        if (!ProcedureCodesData.osrProcedureCodes.contains(code))
+          model.updatedItem(itemId, item => item.copy(fiscalInformation = None, additionalFiscalReferencesData = None))
+        else model
+
+      val step2Model =
+        if (r.isType(DeclarationType.CLEARANCE) && ProcedureCodesData.eicrProcedureCodes.contains(code))
+          step1Model.updatedItem(itemId, item => item.copy(packageInformation = None))
+        else step1Model
+
+      step2Model
+    }
```

Comment:
Could you explain these two `stepModels` please?

Reply:
That was to avoid using var. There are two changes on the models but sequentional.

I'm not sure if without var it's better.

Background you can find in one of my comments above

Reply:
They are a result of a suggestion by Patryk.    Basically this method needs to make (currently) 2 (but will be more in future) passes of the ExportDeclaration model to potentially remove data that is not required due to the current procedure code.

I originally had a var holding the current state of the model which may or may not be updated by each step.

In this version, each step takes the output from the previous step.

Reply:
I numbered the models to hopefully make it more obvious that each step is working on the previous steps output.  But obviously not that obvious.

Reply:
Tech sorting a business issue.
The Questions should be reverted then...

Thank you.

Reply:
I thought about the one approach, but not sure if it will be suitable.

Create a methods to update declaration but which will return Future[ExportsDeclaration], e.g.

```
def updateProcedureCodes(model: ExportDeclaration): Future[ExportsDeclaration] =
  Future.successful(
    if(condition) updateModel
    else model
 )
```

Create a method for every update. and combine it like

`updateProcedurecodes(model).andThen(updateFiscalInformation(_))`

or something like this.

Probably this approach will work somehow, but it will introduce the unnecessary Futures to work on it. It's not a huge deal, because using Future.successful will give us a 100% confidence that it's not something that we need to wait for the result.

Reply:
But with this we will have a different signature for every method. Not return ExportsDeclaration, but `Future[ExportsDeclaration]` which can be more problematic and cause more issues.

I think this is really good to some tech ticket or just tech discussion. How to solve that problem.